App.tsx

```diff
@@ -45,14 +49,100 @@ win.indexedDB.__useShim();
 const njs = require('navcoin-js');
 const P2pPool = require('@aguycalled/bitcore-p2p').Pool;
 
+const currentJSErrorHandler = (e: Error | string, isFatal: boolean) => {
+  let errorMsg: string = '';
+  if (typeof e === 'string') {
+    errorMsg = `Error: ${isFatal ? 'Fatal:' : ''} ${e}
+    Please close the app and start again!`;
+  }
+  if (typeof e === 'object') {
+    errorMsg = `Error: ${isFatal ? 'Fatal:' : ''} ${e.name} ${e.message}
+    Please close the app and start again!`;
+  }
+
+  Alert.alert('Unexpected error occurred', errorMsg, [
+    {
+      text: 'Send report via Email',
+      onPress: () => {
+        sendErrorCrashEmail(e, isFatal);
+        RNRestart.Restart();
+      },
+    },
+    {
+      text: 'Copy Error',
+      onPress: () => {
+        Clipboard.setString(errorMsg);
+        RNRestart.Restart();
+      },
+    },
+    {
+      text: 'Close',
+      onPress: () => {
+        if (!__DEV__) {
+          RNRestart.Restart();
+        }
+      },
+    },
+  ]);
+};
+
+setJSExceptionHandler(currentJSErrorHandler, true);
+
+setNativeExceptionHandler(async errorString => {
+  console.log('setNativeExceptionHandler');
+  console.log(errorString);
+  await AsyncStorage.setItem('crashErrorRecords', errorString);
+});
+
 const App = () => {
   const [theme, setTheme] = React.useState<'light' | 'dark'>('dark');
   const [loaded, setLoaded] = useState(false);
   const {setNjs, setP2pPool} = useNjs();
   const {setWin} = useWin();
 
+  const [crashErrorRecords, setCrashErrorRecords] = useAsyncStorage(
+    'crashErrorRecords',
+    '',
+  );
+
   const [shownWelcome, setShownWelcome] = useState('false');
 
+  const previousNativeErrorHandler = (errorMessage: string) => {
+    Alert.alert(
+      'There was an unexpected error occurred in the previous app launch',
+      'Do you want to send a report to Whisper Team?',
```

Comment:
the Whisper development team

Reply:
There was an unexpected error during the previous session.
Do you want to send a report to the Whisper development team?